cloud/spawn.go

```diff
@@ -203,7 +203,11 @@ func CreateVolume(ctx context.Context, env evergreen.Environment, volume *host.V
 // assumes distro already modified to have one region
 func CheckInstanceTypeValid(ctx context.Context, d distro.Distro, requestedType string, allowedTypes []string) error {
 	if !utility.StringSliceContains(allowedTypes, requestedType) {
-		return errors.New("This instance type has not been allowed by admins")
+		// if it's not in the settings list, check the distro
+		originalInstanceType, ok := d.ProviderSettingsList[0].Lookup("instance_type").StringValueOK()
+		if !ok || originalInstanceType != requestedType {
```

Comment:
maybe `!ok` is a distinct error condition?

Reply:
I don't think so: if there's no instance type then I think we just go forward with the allowedTypes, and this is the correct error message for that still.

Reply:
ah, I was assuming a cached distro with no instance type would be unexpected, like a distro that's been misconfigured.
Is that not the case?

Reply:
Well that is the case for EC2 distros technically? But this doesn't feel like the place to validate that, since it has nothing to do with the user--the useful thing for them is to know that the instance type they want doesn't appear to be available, so I'd rather keep it like this. If a distro got misconfigured it'd be handled somewhere else.

Reply:
I'd argue Go's error handling story pushes us to handle errors in unlikely places, but I will not push further on this issue.
🤷‍♂️